nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/schema/inference/InferSchemaAccessStrategy.java

```diff
@@ -43,7 +43,7 @@ public InferSchemaAccessStrategy(final RecordSourceFactory<T> recordSourceFactor
     public RecordSchema getSchema(final Map<String, String> variables, final InputStream contentStream, final RecordSchema readSchema) throws IOException {
         // We expect to be able to mark/reset any length because we expect that the underlying stream here will be a ContentClaimInputStream, which is able to
         // re-read the content regardless of how much data is read.
-        contentStream.mark(10_000_000);
+        contentStream.mark(Integer.MAX_VALUE);
```

Comment:
I think we want to keep this at a smaller value. The general contract of `InputStream` says that when `mark` is called, the stream must remember at least that many bytes but is free to remember more. As the comment above explains, the general expectation is that the stream will be of type ContentClaimInputStream. In that case, the value passed to `mark` is ignored because it can always roll back to the beginning of the stream (by re-reading the file under the hood).

But keeping the 10 MB limit (or even a 1 MB limit would probably be okay) means that if the caller does wrap the InputStream in a BufferedInputStream, then we have the chance remember this amount of data. If it's changed to Integer.MAX_VALUE, that bound is basically removed, which can lead to OutOfMemoryError very easily, and that should be avoided at almost any cost as the entire JVM can become defunct.

Reply:
I see your point and share your concern.
Then I'll revert this change and also remove the newly added test case and keep only the `BufferedInputStream` wrapping removal.
I also checked all the other references to `org.apache.nifi.serialization.SchemaRegistryService#getSchema()` and no other `BufferedInputStream` wrapping occurs, at least not directly before the `getSchema()` call.